lib/charms/postgresql_k8s/v1/postgresql.py

```diff
@@ -257,74 +272,126 @@ def create_database(
             logger.error(f"Failed to create database: {e}")
             raise PostgreSQLCreateDatabaseError() from e
 
-        # Enable preset extensions
-        self.enable_disable_extensions(dict.fromkeys(plugins, True), database)
-
     def create_user(
         self,
         user: str,
         password: Optional[str] = None,
-        admin: bool = False,
-        extra_user_roles: Optional[List[str]] = None,
+        roles: Optional[List[str]] = None,
     ) -> None:
         """Creates a database user.
 
         Args:
             user: user to be created.
             password: password to be assigned to the user.
             admin: whether the user should have additional admin privileges.
-            extra_user_roles: additional privileges and/or roles to be assigned to the user.
+            roles: roles to be assigned to the user
         """
+
+        superuser_enabled, createdb_enabled, createrole_enabled = False, False, False
+        if "admin" in roles:
+            createdb_enabled, createrole_enabled = True, True
+            roles.append(ROLE_DML)
+            roles.append(ROLE_INSTANCE_ADMIN)
+            roles.remove("admin")
+
+        if "createdb" in roles:
+            createdb_enabled = True
+            roles.remove("createdb")
+
+        if "createrole" in roles:
+            createrole_enabled = True
+            roles.remove("createrole")
+
+        if "superuser" in roles:
+            superuser_enabled = True
+            roles.remove("superuser")
+
```

Comment:
Are createdb/createrole necessary to be changed in this PR?

Reply:
i believe that the implementation is different but functionality is actually identical. one can specify `extra-user-roles=createdb,createrole,admin,superuser` as they could before. we just process the requested roles using conditions like this instead of creating a whitelist of accepted attributes/privileges